payments/management/commands/generate_discount_code.py

```diff
@@ -0,0 +1,164 @@
+"""
+This module contains the management command to generate discount codes.
+"""
+
+import csv
+
+from django.core.management import BaseCommand
+
+from payments.api import generate_discount_code
+
+
+class Command(BaseCommand):
+    """
+    Generates discount codes.
+    An example usage of this command:
+    python manage.py generate_discount_code --payment-type marketing \
+    --amount 10 --count 5 --one-time
+    """
+
+    help = "Generates discount codes."
+
+    def add_arguments(self, parser) -> None:
+        """
+        Add command line arguments to the parser.
+        """
+        parser.add_argument(
+            "--prefix",
+            type=str,
+            help="The prefix to use for the codes. (Maximum length 13 characters)",
+        )
+
+        parser.add_argument(
+            "--expires",
+            type=str,
+            help=(
+                "Optional expiration date for the code, "
+                "in ISO-8601 (YYYY-MM-DD) format."
+            ),
+        )
+
+        parser.add_argument(
+            "--activates",
+            type=str,
+            help=(
+                "Optional activation date for the code, "
+                "in ISO-8601 (YYYY-MM-DD) format."
+            ),
+        )
+
+        parser.add_argument(
+            "--discount-type",
+            type=str,
+            help=(
+                "Sets the discount type (dollars-off, percent-off, fixed-price; "
+                "default percent-off)"
+            ),
+            default="dollars-off",
+        )
+
+        parser.add_argument(
+            "--payment-type",
+            type=str,
+            help=(
+                "Sets the payment type (marketing, sales, financial-assistance, "
+                "customer-support, staff)"
+            ),
+            required=True,
+        )
+
+        parser.add_argument(
+            "--amount",
+            type=str,
+            nargs="?",
+            help="Sets the discount amount",
+            required=True,
+        )
+
+        parser.add_argument(
```

Comment:
Can this not be required if the user specifies codes on the command line? It should either require `--codes` and `--prefix` or a list of codes to create.